File: src/domain/entities/category.rs
use uuid::Uuid;

#[derive(Debug, Clone)]
pub struct Category {
    id: Uuid,
    name: String,
    description: Option<String>,
    color: String,
}

impl Category {
    pub fn new(id: Uuid, name: String, description: Option<String>, color: String) -> Self {
        Category {
            id,
            name,
            description,
            color,
        }
    }

    pub fn id(&self) -> Uuid {
        self.id
    }

    pub fn name(&self) -> &str {
        &self.name
    }

    pub fn description(&self) -> Option<&str> {
        self.description.as_deref()
    }

    pub fn color(&self) -> &str {
        &self.color
    }
}

#[cfg(test)]
mod category_tests {
    use super::*;

    #[test]
    fn test_new_category() {
        let id = Uuid::new_v4();
        let name = "Test Category".to_string();
        let description = Some("This is a test category".to_string());
        let color = "#FF0000".to_string();

        let category = Category::new(id, name, description, color);

        assert_eq!(category.id(), id);
        assert_eq!(category.name(), "Test Category");
        assert_eq!(category.description(), Some("This is a test category"));
        assert_eq!(category.color(), "#FF0000");
    }
}

File: app/lib/domain/entities/category.dart
import 'package:equatable/equatable.dart';

class Category extends Equatable {
  final String id;
  final String name;
  final String color;
  final String? description;
  final DateTime createdAt;
  final DateTime updatedAt;

  const Category({
    required this.id,
    required this.name,
    required this.color,
    this.description,
    required this.createdAt,
    required this.updatedAt,
  });

  Category copyWith({
    String? id,
    String? name,
    String? color,
    String? description,
    DateTime? createdAt,
    DateTime? updatedAt,
  }) {
    return Category(
      id: id ?? this.id,
      name: name ?? this.name,
      color: color ?? this.color,
      description: description ?? this.description,
      createdAt: createdAt ?? this.createdAt,
      updatedAt: updatedAt ?? this.updatedAt,
    );
  }

  @override
  String toString() {
    return 'Category(id: $id, name: $name, color: $color, description: $description, createdAt: $createdAt, updatedAt: $updatedAt)';
  }

  @override
  List<Object?> get props => [
    id,
    name,
    color,
    description,
    createdAt,
    updatedAt,
  ];
}
